custom/icds_reports/reports/take_home_ration.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import unicode_literals
+
+from corehq.apps.locations.models import SQLLocation
+from custom.icds_reports.utils import india_now, DATA_NOT_ENTERED
+from custom.icds_reports.models.views import TakeHomeRationMonthly
+
+
+class TakeHomeRationExport(object):
+    title = 'Take Home Ration'
+
+    def __init__(self, location, month, loc_level=0, beta=False):
+        self.location = location
+        self.loc_level = loc_level
+        self.month = month
+        self.beta = beta
+
+    def get_excel_data(self):
+
+        def _format_infrastructure_data(data):
+            return data if data is not None else DATA_NOT_ENTERED
+
+        if self.loc_level == 4:
+            data = TakeHomeRationMonthly.objects.filter(
+                supervisor_id=self.location, month=self.month, aggregation_level=5
+            ).order_by('awc_name')
+        elif self.loc_level == 3:
+            data = TakeHomeRationMonthly.objects.filter(
+                block_id=self.location, month=self.month, aggregation_level=5
+            ).order_by('supervisor_name', 'awc_name')
+        elif self.loc_level == 2:
+            data = TakeHomeRationMonthly.objects.filter(
+                district_id=self.location, month=self.month, aggregation_level=5
+            ).order_by('block_name', 'supervisor_name', 'awc_name')
+        elif self.loc_level == 1:
+            data = TakeHomeRationMonthly.objects.filter(
+                state_id=self.location, month=self.month, aggregation_level=5
+            ).order_by('district_name', 'block_name', 'supervisor_name', 'awc_name')
+        else:
+            data = TakeHomeRationMonthly.objects.filter(
+                month=self.month, aggregation_level=5
+            ).order_by('state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name')
```

Comment:
These filters are mostly the same. I think that it may be clearer what the if/else statements are controlling if the filters were constructed like:

```python
filters = {"month": self.month, "aggregation_level": 5}
if self.loc_level == 4:
   filters["supervisor_id"] = self.location_id
   order_by = ('awc_name',)
elif self.loc_level == 3:
   filters["block_id"] = self.location_id
   order_by = ("supervisor_name", "awc_name")
...

data = TakeHomeRationMonthly.objects.filter(**filters).order_by(*order_by)
```

Reply:
ohh yeah right.....Done changing